src/bundles/files.js

```diff
@@ -119,26 +119,35 @@ function readAsBuffer (file) {
   })
 }
 
-bundle.doFilesWrite = (root, files) => ({dispatch, getIpfs, store}) => {
+bundle.doFilesWrite = (root, files) => async ({dispatch, getIpfs, store}) => {
   dispatch({ type: 'FILES_WRITE_STARTED' })
 
-  let promises = []
+  try {
+    let promises = []
 
-  for (const file of files) {
-    promises.push(readAsBuffer(file))
-  }
+    for (const file of files) {
+      promises.push(readAsBuffer(file))
+    }
+
+    files = await Promise.all(promises)
+
+    for (const file of files) {
+      const path = join(root, dirname(file.name))
+      await getIpfs().files.mkdir(path, { parents: true })
+    }
 
-  return Promise.all(promises).then(files => {
-    return Promise.all(files.map((file) => {
-      const target = join(root, file.name)
-      return getIpfs().files.write(target, file.content, { create: true })
+    await Promise.all(files.map(async file => {
+      const res = await getIpfs().add([file.content])
```

Comment:
I wonder if we should disable pinning when `add`ing for MFS. 
As long as files are in MFS, they won't be GC'd.
If we dont pass `{pin: false}` to `add` then removing file from MFS won't remove it from node.  
Is that a [bug or feature](https://ipfs.io/ipfs/QmVnLkRxWWv2xiPKeQy6re1zCepzX7YXUiq5fESkd1j8bC/bug_or_feature.jpg) ?
@olizilla  any thoughts on this?

Reply:
I'm adding that option then.

About the progress of uploading: don't worry. Won't be doing in this PR but it is on Invision how it is supposed to be.

Reply:
Done @lidel